### PR TITLE
Feat collection bulk ops

### PR DESCRIPTION
**Pull request by Lasse-numerous**
This pull request introduces new bulk_download and bulk_upload functionalities to the Python SDK, allowing for efficient management of entire collection hierarchies.
Key Features:
bulk_download: Recursively downloads a full collection from Numerous to the local filesystem.
Files are stored as binary data.
Documents are serialized to .json files.
The remote collection structure is mirrored locally.
bulk_upload: Recursively uploads a local directory to a Numerous collection.
Creates nested collections from local subdirectories.
Uploads .json files as Numerous documents.
Supports a .numerous-ignore file (using gitignore-style patterns) to exclude specific files or directories from the upload.
Implementation Details:
Core Logic: Implemented in the new numerous/collections/bulk/ module.
Dependencies: Uses the pathspec library for gitignore-style pattern matching.
Testing: Includes a comprehensive test suite (test_collection_bulk.py) with 70+ tests covering happy paths, error conditions, and edge cases.
Documentation:
Adds a new collection_bulk_operations.md page to the SDK documentation with a detailed guide, API reference, and best practices.
Includes three new examples: collection_bulk_download.py, collection_bulk_upload.py, and collection_bulk_advanced.py.

### CODE REVIEW

File: docs/collection_bulk_operations.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+-   **Upload** a local directory structure (containing files and documents) into a Numerous collection, automatically creating nested collections as needed.
+
+This is particularly useful for:
+
+-   Working with large datasets locally.
+-   Batch-publishing new content or updates.
+-   Backing up collection data.
+-   Migrating data between local storage and Numerous.
+
+Files are transferred as binary data, preserving their content. Documents are typically represented as JSON files locally and are stored as structured data in Numerous.
```
Comment (mikhailNumerous) [inline docs/collection_bulk_operations.md:19]: @LasseNT, just an idea to make files and documents distinction more specific. Maybe, we can use some suffix, e.g. `.numerous-doc.json` instead of `json` extension. Or even add the suffix as an optional parameter of `bulk_download` and `bulk_upload`. E.g.:
``` python
def bulk_download(
    collection_ref: CollectionReference,
    local_base_path: Path = Path("collections"),
    document_suffix: str | None = None,
) -> None:

def bulk_upload(
    collection_ref: CollectionReference,
    local_base_path: Path = Path("collections"),
    ignore_file_name: str = DEFAULT_IGNORE_FILE,
    document_suffix: str | None = None,
) -> None:
```

File: docs/collection_bulk_operations.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    -   If an error occurs (e.g., a single file fails to download/upload, a JSON document is malformed), an error message will be logged, and the operation will attempt to continue with the next items.
+-   Invalid Keys: If a local file or directory name is invalid for use as a Numerous collection or entity key during `bulk_upload` (e.g., contains forbidden characters), a warning will be logged, and that specific item will be skipped.
+
+Consult the SDK's general error handling documentation for more details on common exception types like `GraphQLClientError`.
+
+## Best Practices
+
+-   **Test with a Sample Collection**: Before running bulk operations on critical data, test your setup and ignore patterns with a smaller, non-critical sample collection.
+-   **Monitor Logs**: The bulk operations log information about their progress and any errors encountered. Check these logs to verify the process.
+-   **Understand Overwrites**: Both download and upload operations will overwrite existing data (local files for download, remote Numerous entities for upload) without prompting. Ensure this is the desired behavior.
```
Comment (mikhailNumerous) [inline docs/collection_bulk_operations.md:164]: It worth to mention in the docs what is the expected behavior in edge case if a file and a document with the same (without extension / suffix) name exist.
E.g.: `config.json` has already been manually uploaded and there is also a local file to be uploaded with `bulk_upload` as a document. So on `bulk_download` there are should be a discussed and expected order of such overwrite.

File: python/src/numerous/collections/bulk/helpers.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    If the directory structure does not exist, it will be created.
+
+    Args:
+        path: The path to the directory to create.
+
+    """
+    path.mkdir(parents=True, exist_ok=True)
+
+
+def _get_nested_collection_path(current_local_base: Path, collection_key: str) -> Path:
```
Comment (mikhailNumerous) [inline python/src/numerous/collections/bulk/helpers.py:31]: The function is only tested and not used elsewhere.

File: docs/collection_bulk_operations.md (hunk 1 above)
Comment (Lasse-numerous) [reply docs/collection_bulk_operations.md]: Yes  - great idea. I've updated using "collection-doc" to identify docs.

File: docs/collection_bulk_operations.md (hunk 2 above)
Comment (Lasse-numerous) [reply docs/collection_bulk_operations.md:247]: @mikhailNumerous After scoping collection docs with "collection-doc.json" this is no longer an issue, right?

Comment (mikhailNumerous) [reply docs/collection_bulk_operations.md:247]: Yes, it already has been an edge case and it just looks as a detail to be documented rather than issue.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -37,7 +38,8 @@ repos:
         name: golangci-lint
         language: system
         stages: [pre-commit, pre-push]
-        entry: bash -c 'golangci-lint run --allow-parallel-runners'
```
Comment (mikhailNumerous) [inline .pre-commit-config.yaml:40]: Just curious if there are any issues with `--allow-parallel-runners`.

File: docs/collection_bulk_operations.md (hunk 2 above)
Comment (Lasse-numerous) [reply docs/collection_bulk_operations.md:266]: @mikhailNumerous  With addition of the collection-doc.json suffix config.json would be treated as a file - whereas config.collection-doc.json would be a document. I have added to docs that these will be treated as separate entities now.

File: .pre-commit-config.yaml (hunk 4 above)
Comment (Lasse-numerous) [reply .pre-commit-config.yaml:40]: @mikhailNumerous I had some issues with golanci-lint being very slow and I experimented with removing the allow-parallel-runners flag. I have reintroduced it again.
